### PR TITLE
subscribe into master

### PR DESCRIPTION
**Pull request by Shalifar**

### CODE REVIEW

File: src/AppBundle/Controller/SubscribeController.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     *
+     * @param Request
+     * @return \Symfony\Component\HttpFoundation\Request
+     */
+    public function showForm(Request $request)
+    {
+        $subscriber = new Subscriber();
+
+        $form = $this->createFormBuilder($subscriber)
+            ->add('email', TextType::class)
```
Comment (asjustas) [inline src/AppBundle/Controller/SubscribeController.php]: http://symfony.com/doc/current/reference/forms/types/email.html

File: src/AppBundle/Controller/SubscribeController.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            ->getForm();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+
+            $subscriber = $form->getData();
+
+            if (!filter_var($subscriber->getEmail(), FILTER_VALIDATE_EMAIL)) {
```
Comment (asjustas) [inline src/AppBundle/Controller/SubscribeController.php]: https://symfony.com/doc/current/reference/constraints/Email.html

File: src/AppBundle/Controller/SubscribeController.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                return $this->render('error/error.html.twig', [
+                    'errorMsg' => 'Invalid e-mail!',
+                ]);
+            }
+
+            //email check
+            $users = $em->getRepository('AppBundle:Subscriber')->findBy(['email' => $subscriber->getEmail()]);
```
Comment (asjustas) [inline src/AppBundle/Controller/SubscribeController.php]: findOneBy negali naudoti?

File: src/AppBundle/Controller/SubscribeController.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            if (isset($users[0])) {
+                return $this->render('error/error.html.twig', [
+                    'errorMsg' => 'Such e-mail already subscribed!',
+                ]);
+            }
+
+            //keyName check
+            do {
```
Comment (asjustas) [inline src/AppBundle/Controller/SubscribeController.php:52]: negeriau naudoti http://php.net/uniqid ar kazka panasaus?

Comment (Shalifar) [reply src/AppBundle/Controller/SubscribeController.php:52]: o koks skirtumas? uniqueID vis tiek negarantuoja unikaluma

**Pull request merged**